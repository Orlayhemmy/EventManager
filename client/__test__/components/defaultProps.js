export const props = {
  userState: {
    user: {
      isAdmin: true
    },
    isAuth: true
  },
  location: {
    pathname: '/'
  },
  center: {
    status: '',
    message: ''
  }
};

export const center = {
  status: 201,
  message: ''
};

export const bookedEvent = [
  { id: 1, eventTitle: 'Pizza Party', bookedDate: [], isApproved: true },
  { id: 2, eventTitle: 'Wedding Ceremony', bookedDate: [], isApproved: true  },
  { id: 3, eventTitle: 'Seminar', bookedDate: [], isApproved: false  }
];

export const logout = jest.fn();
export const onChange = jest.fn();
