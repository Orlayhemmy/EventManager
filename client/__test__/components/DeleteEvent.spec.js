import React from 'react';
import DeleteContent from '../../components/CenterDetails/Template/Content/deleteEvent';
import { onChange } from './defaultProps';

describe.only('Approve Event Component', () => {
  const onApprove = jest.fn();
  const state = {
    reason: '',
    suggestions: ''
  };
  it('should render component without error', () => {
    const wrapper = renderer
      .create(<DeleteContent onChange={onChange} onApprove={onApprove} state={state} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
