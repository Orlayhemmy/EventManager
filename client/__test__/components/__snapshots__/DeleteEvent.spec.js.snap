// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approve Event Component should render component without error 1`] = `
<div
  className="form-inner"
>
  <span
    className="help-block"
  >
    Are sure you want to delete event?
  </span>
  <br />
  <br />
  <div
    className="form-group"
  >
    <textarea
      className="form-control"
      id="reason"
      onChange={[MockFunction]}
      placeholder="Give reasons for disapproving this event"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <textarea
      className="form-control"
      id="suggestion"
      onChange={[MockFunction]}
      placeholder="Suggestions"
      value={undefined}
    />
  </div>
  <i
    className="fa fa-trash red"
    id="disapprove"
    onClick={[MockFunction]}
  />
  <i
    className="fa fa-save green"
  />
  <br />
  <span>
    <br />
    Yes
  </span>
  <span>
    <br />
    No
  </span>
</div>
`;
